Reset no longer needed on staking & fix underflow

danielattilasimon opened the pull request:
Address remaining TODO in `_increaseUserVoteTrackers()` by eliminating reset requirement.
Fixes #108.

Fix possible underflow in `withdrawLQTY()`.
Fixes #109.

bingen reviewed (approved):
Looks good!
I think there are some other places where we may update tests to the fact that we don’t need resets on deposit/withdrawal anymore, like:

https://github.com/liquity/V2-gov/blob/main/test/recon/targets/GovernanceTargets.sol#L110
https://github.com/liquity/V2-gov/blob/main/test/recon/targets/GovernanceTargets.sol#L283

But happy to merge this as is, we can fix that later if it’s more convenient.

danielattilasimon commented on src/Governance.sol:
```diff
@@ -203,7 +196,6 @@ contract Governance is MultiDelegateCall, UserProxyFactory, ReentrancyGuard, Own
     function withdrawLQTY(uint256 _lqtyAmount, bool _doSendRewards, address _recipient) public nonReentrant {
         // check that user has reset before changing lqty balance
```
Oops 😅

danielattilasimon commented:
I'd fix the invariant tests in a separate PR later, as there are already some broken ones after the voting power refactor.

danielattilasimon commented:
On the other hand, I dug up the test case I wrote that was demonstrating the bug that was the whole reason for the reset requirement, updated it, and added it to the test suite.

bingen commented:
Awesome, thanks!!
